app/models/state_file_base_intake.rb

```diff
@@ -190,4 +192,32 @@ def increment_failed_attempts
       lock_access! unless access_locked?
     end
   end
+
+  def dependents_eligible_for_child_tax_credit
+    dependents = []
+    self.direct_file_data&.parsed_xml&.css('DependentDetail').each do |dependent|
+      if ELIGIBLE_CTC.in? dependent.to_s
+        first_name = dependent.at('DependentFirstNm').text
+        last_name = dependent.at('DependentLastNm').text
+        relationship = dependent.at('DependentRelationshipCd').text
+        dependents << StateFileDependent.where(
+          first_name: first_name, last_name: last_name, relationship: relationship
+        ).take
+      end
+    end
+    dependents.compact
+  end
+
+  def dependents_eligible_for_eitc
+    dependents = []
+    self.direct_file_data&.parsed_xml&.css('IRS1040ScheduleEIC QualifyingChildInformation').each do |dependent|
+      first_name = dependent.at('PersonFirstNm').text
+      last_name = dependent.at('PersonLastNm').text
+      relationship = dependent.at('ChildRelationshipCd').text
+      dependents << StateFileDependent.where(
+        first_name: first_name, last_name: last_name, relationship: relationship
+      ).take
+    end
+    dependents.compact
+  end
```

Comment:
This code feels duplicative and less clear to me than [the current code in direct_file_data](https://github.com/codeforamerica/vita-min/blob/main/app/models/direct_file_data.rb#L572-L592) that currently handles parsing dependent xml and determining whether they are eligible for eic. We have a column named `eic_qualifying` that should already be storing whether the dependent is eligible for eic or not. Is there a reason that these model methods should parse the xml rather than use their existing database column? I think that if we can store `eic_qualifying` in the database (perhaps also adding `ctc_qualifying` column?), we should use that.

Reply:
Yeah, that is something I thought about but the direct file data XML did not match the DB column, but whether that mechanism is working properly or nor was not something I dug in extensively.